core/services/ping/pingmanager.py

```diff
@@ -1,38 +1,69 @@
+import asyncio
 import logging
-from typing import Dict, List
+from typing import Any, Dict, List
 
 from serial import Serial
 
 from ping1d_driver import Ping1DDriver
 from ping360_driver import Ping360Driver
+from ping360_ethernet_driver import Ping360EthernetDriver
 from pingdriver import PingDriver
-from pingutils import PingDeviceDescriptor, PingType
+from pingutils import PingDeviceDescriptor, PingType, udp_port_is_in_use
 
 
 class PingManager:
     def __init__(self) -> None:
         self.drivers: Dict[PingDeviceDescriptor, PingDriver] = {}
-        self.ping1d_current_port: int = 9090
-        self.ping360_current_port: int = 9092
+        self.ping1d_base_port: int = 9090
+        self.ping360_base_port: int = 9092
 
     def stop_driver_at_port(self, port: Serial) -> None:
         """Stops the driver instance running for port "port" """
-        ping_at_port = list(filter(lambda ping: ping.port == port, self.drivers.keys()))
+        ping_at_port = [ping for ping in self.drivers if ping.port == port]
         if ping_at_port:
             self.drivers[ping_at_port[0]].stop()
+            del self.drivers[ping_at_port[0]]
 
-    def launch_driver_instance(self, ping: PingDeviceDescriptor) -> None:
-        """Launches a new driver instance for the PingDeviceDescriptor "ping" """
+    async def register_ethernet_ping360(self, ping: PingDeviceDescriptor) -> None:
+        if ping not in self.drivers:
+            self.drivers[ping] = Ping360EthernetDriver(ping)
+
+    async def find_next_port(self, base_port: int, step: int) -> int:
+        """
+        Finds the next unused UDP port.
+        Starts at 'base_port' and increments/decrements by 'step'.
+        """
+        port = base_port
+        while udp_port_is_in_use(port):
+            port += step
+            await asyncio.sleep(0.1)
+        return port
+
+    async def launch_driver_instance(self, ping: PingDeviceDescriptor) -> None:
+        """Launches a new driver instance for the PingDeviceDescriptor "ping"."""
         driver: PingDriver
         if ping.ping_type == PingType.PING1D:
             logging.info("Launching ping1d driver")
-            driver = Ping1DDriver(ping, self.ping1d_current_port)
+            port = await self.find_next_port(self.ping1d_base_port, step=-1)
+            driver = Ping1DDriver(ping, port)
         elif ping.ping_type == PingType.PING360:
             logging.info("Launching ping360 driver")
-            driver = Ping360Driver(ping, self.ping360_current_port)
+            port = await self.find_next_port(self.ping360_base_port, step=+1)
```

Comment:
How the +1 and -1 logic will work with future sensors ?
As pointed in the idea over: https://github.com/bluerobotics/BlueOS-docker/pull/1111/files#r941320562

Reply:
the only reason we needed `step` here was because they 9092 and 9090 are way to close, and the sensors ports would collide after we have two ping1Ds.

The step could be removed now that we have a udp_port_is_in_use(). but I would rather have a somewhat logical order, while we don't have proper discovery set up.